### DISCUSSION TITLE
Routing without time windows gives non homogenous results

### DISCUSSION THREAD

**Discussion by lleihpi** [google/or-tools#3500] [Routing (and legacy CP) questions] [answered]
I just wanted to know if this is a regular behavior for the solver. I am solving a case with 325 orders and 6 vehicles with 75 orders capacity. All of the vehicles start and end at the same point. The routes given by the solver with GUIDED LOCAL SEARCH and 60 seconds is way too mixed between vehicles resulting in them having visually bad routes. I have tried increasing time to 360 seconds but the result remains exactly the same which makes it seem this is the best result the tool can find. Is this behavior expected? I expect this with time windows or constrained excercises but one with just a capacity constraint seems wrong that it doesn't give a better result.

We are using this algorithm in other use cases but this is the first time I am not sure of the solution. Any feedback is appreciated. 

<img width="608" alt="Screen Shot 2022-10-17 at 9 29 49 AM" src="https://user-images.githubusercontent.com/20404122/196219720-84da2849-3241-434a-b74d-e02c6334c5d8.png">

**Comment by lleihpi** [comment]
Just read the comments on this thread https://github.com/google/or-tools/discussions/3414 , it seems this is a normal behavior.

**Comment by jmarca** [comment, marked as answer]
Well, I wrote a comment on that other thread and was going to reply to this one with
similar!

What I'd add is that there are ways to balance routes if you want.
For example, minimizing the maximum difference between the shortest
and longest route (search for makespan minimization) in the examples.

Visually inspecting the routes is a good thing to do, and it can
reveal that you've missed something important in your model.  But by
the same token, you can't be too rigid about forcing the solver to use
your old heuristics.

James


On Mon, Oct 17, 2022 at 08:38:48AM -0700, lleihpi wrote:
> Just read the comments on this thread https://github.com/google/or-tools/discussions/3414 , it seems this is a normal behavior.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/google/or-tools/discussions/3500#discussioncomment-3897832
> You are receiving this because you are subscribed to this thread.
>
> Message ID: ***@***.***>

--

James E. Marca
Activimetrics LLC

**Reply by lleihpi** [reply]
Thanks James! It is just that sometimes its hard to explain this cases to users so I wanted to have the right arguments. Thanks for your  answer.

**Comment by jmarca** [comment]
Ah yes, working with actual users can be tricky.  Remember that the
first rule should be that the user is always right, and you have to
listen very carefully to their complaints/comments.  The "inefficient"
clustered routes might have hidden efficiencies that the solver is
missing, such as the drivers becoming familiar with a fixed area, etc.

But by the same token, the whole point of using a computer is that
formulating these routes by hand is too hard.

Good luck,

James
